dawn/src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -292,6 +294,7 @@ DawnCompiler::optimize(std::map<std::string, std::shared_ptr<iir::StencilInstant
     // since this can change the scope of temporaries ...
     optimizer.pushBackPass<PassTemporaryType>();
     optimizer.pushBackPass<PassLocalVarType>();
+    optimizer.pushBackPass<PassRemoveScalars>();
```

Comment:
how can we generate new scalars with merging?

Reply:
If you never assign an expression with dimensions to a temporary, after merging stages it will become a local variable which is flagged as scalar by `PassLocalVarType`.

Reply:
got it